java-checks/src/main/java/org/sonar/java/checks/helpers/QuickFixHelper.java

```diff
@@ -118,4 +136,99 @@ public static String contentForTree(Tree tree, JavaFileScannerContext context) {
 
     return sb.toString();
   }
+
+  public static Optional<JavaTextEdit> newImportEdit(String requiredType, Tree base) {
+    if (!requiresImportOf(requiredType, base)) {
+      return Optional.empty();
+    }
+
+    CompilationUnitTree cut = ((JavaTree) base).root();
```

Comment:
The first step of the method `requiresImportOf` is to compute this value, you could give this `cut` as argument, and avoid computing it a second time.

Reply:
it is not "computed" per se, as it's always available. But I agree it's not needed.